### PR TITLE
COMMERCE-4977 Create new Admin Shipment API

### PR DESCRIPTION
**Pull request by riccardo-alberti**

### CODE REVIEW

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentResourceImpl.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+				commerceOrder.getShippingAddressId(),
+				commerceOrder.getCommerceShippingMethodId(),
+				commerceOrder.getShippingOptionName(),
+				_serviceContextHelper.getServiceContext(contextUser));
+
+		return _toShipment(commerceShipment.getCommerceShipmentId());
+	}
+
+	@Override
+	public Shipment postShipmentDeliver(Long shipmentId) throws Exception {
```
Comment (dsitu) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentResourceImpl.java:139]:
```suggestion
	public Shipment postShipmentStatusDelivered(Long shipmentId) throws Exception {
```

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentResourceImpl.java [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+		return _toShipment(
+			_commerceShipmentService.updateStatus(
+				shipmentId,
+				CommerceShipmentConstants.SHIPMENT_STATUS_DELIVERED));
+	}
+
+	@Override
+	public Shipment postShipmentFinishProcessing(Long shipmentId)
```
Comment (dsitu) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentResourceImpl.java:147]:
```suggestion
	public Shipment postShipmentStatusFinishedProcessing(Long shipmentId)
```

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentResourceImpl.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+		throws Exception {
+
+		return _toShipment(
+			_commerceShipmentService.updateStatus(
+				shipmentId,
+				CommerceShipmentConstants.SHIPMENT_STATUS_READY_TO_BE_SHIPPED));
+	}
+
+	@Override
+	public Shipment postShipmentShip(Long shipmentId) throws Exception {
```
Comment (dsitu) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentResourceImpl.java:157]:
```suggestion
	public Shipment postShipmentStatusShipped(Long shipmentId) throws Exception {
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentItemResourceImpl.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+			Long shipmentItemId, ShipmentItem shipmentItem)
+		throws Exception {
+
+		CommerceShipmentItem commerceShipmentItem =
+			_commerceShipmentItemService.getCommerceShipmentItem(
+				shipmentItemId);
+
+		_commerceShipmentItemService.updateCommerceShipmentItem(
+			shipmentItemId,
+			GetterUtil.get(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentItemResourceImpl.java:95]: is it allowed to change warehouse ?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/util/v1_0/ShippingAddressUtil.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+import com.liferay.headless.commerce.core.util.ServiceContextHelper;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+/**
+ * @author Riccardo Alberti
+ */
+public class ShippingAddressUtil {
+
+	public static CommerceShipment upsertShippingAddress(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/util/v1_0/ShippingAddressUtil.java:35]: there is no insert method in this upsert

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:foward

**Pull request closed without merging**